admin/load.php

```diff
@@ -0,0 +1,295 @@
+		$sections[ $module_section ]['added'] = true;
+
+		add_settings_field(
+			$module_slug,
+			$module_data['name'],
+			function() use ( $module_slug, $module_data, $module_settings ) {
+				perflab_render_modules_page_field( $module_slug, $module_data, $module_settings );
+			},
+			PERFLAB_MODULES_SCREEN,
+			$module_section
+		);
+	}
+
+	// Remove all sections for which there are no modules.
+	foreach ( $sections as $section_slug => $section_data ) {
+		if ( empty( $section_data['added'] ) ) {
+			unset( $wp_settings_sections[ PERFLAB_MODULES_SCREEN ][ $section_slug ] );
+		}
+	}
+}
+
+/**
+ * Renders the modules page.
+ *
+ * @since 1.0.0
+ */
+function perflab_render_modules_page() {
+	?>
+	<div class="wrap">
+		<h1>
+			<?php esc_html_e( 'Performance Modules', 'performance-lab' ); ?>
+		</h1>
+
+		<form action="options.php" method="post" novalidate="novalidate">
+			<?php settings_fields( PERFLAB_MODULES_SCREEN ); ?>
+			<?php do_settings_sections( PERFLAB_MODULES_SCREEN ); ?>
+			<?php submit_button(); ?>
+		</form>
+	</div>
+	<?php
+}
+
+/**
+ * Renders fields for a given module on the modules page.
+ *
+ * @since 1.0.0
+ *
+ * @param string $module_slug     Slug of the module.
+ * @param array  $module_data     Associative array of the module's parsed data.
+ * @param array  $module_settings Associative array of the module's current settings.
+ */
+function perflab_render_modules_page_field( $module_slug, $module_data, $module_settings ) {
+	$base_id   = sprintf( 'module_%s', $module_slug );
+	$base_name = sprintf( '%1$s[%2$s]', PERFLAB_MODULES_SETTING, $module_slug );
+	$enabled   = isset( $module_settings['enabled'] ) && $module_settings['enabled'];
+	?>
+	<fieldset>
+		<legend class="screen-reader-text">
+			<?php echo esc_html( $module_data['name'] ); ?>
+		</legend>
+		<label for="<?php echo esc_attr( "{$base_id}_enabled" ); ?>">
+			<input type="checkbox" id="<?php echo esc_attr( "{$base_id}_enabled" ); ?>" name="<?php echo esc_attr( "{$base_name}[enabled]" ); ?>" aria-describedby="<?php echo esc_attr( "{$base_id}_description" ); ?>" value="1"<?php checked( $enabled ); ?>>
+			<?php
+			if ( $module_data['experimental'] ) {
+				printf(
+					/* translators: %s: module name */
+					__( 'Enable %s <strong>(experimental)</strong>?', 'performance-lab' ),
+					esc_html( $module_data['name'] )
+				);
+			} else {
+				printf(
+					/* translators: %s: module name */
+					__( 'Enable %s?', 'performance-lab' ),
+					esc_html( $module_data['name'] )
+				);
+			}
+			?>
+		</label>
+		<p id="<?php echo esc_attr( "{$base_id}_description" ); ?>" class="description">
+			<?php echo esc_html( $module_data['description'] ); ?>
+		</p>
+	</fieldset>
+	<?php
+}
+
+/**
+ * Gets all available focus areas.
+ *
+ * @since 1.0.0
+ *
+ * @return array Associative array of focus area data, keyed by focus area slug. Fields for every focus area include
+ *               'name'.
+ */
+function perflab_get_focus_areas() {
+	return array(
```

Comment:
I wonder if we shouldn't dynamically get this dynamically from the modules header. On one hand it is more dynamic on the other it can easily result to duplicates due to typo or else. Thoughts

Also, I do think that we need an additional focus area for the "Infrastructure/API" modules.

Reply:
I think having these defined in a central location helps providing guidance on what to specify in the module header, and it also allows us to provide any extra information about a focus area (e.g. if in the future we wanted to add descriptions for each focus to the settings screen).

Adding a new focus area would be straightforward, it would only require a PR to expand this array (and it already requires a workflow anyway, as it will need a new GitHub label to be created as well).

Regarding something for "Infrastructure/API", I think that goes into the area of dependencies, or modules that go beyond a single focus. Maybe better to open a separate issue to discuss that, since it's a larger topic? Also it might be hard to evaluate this early as we aren't yet in a situation where it would be needed.